runtime/vm/FFITypeHelpers.hpp

```diff
@@ -248,6 +248,89 @@ class FFITypeHelpers
 		return n;
 	}
 
+	/**
+	 * @brief obtain the ffi_type from the layout symbol (the starting letter of the layout type)
+	 * @param layoutSymb[in] The layout symbol describing the type of the layout
+	 * @return The pointer to the ffi_type corresponding to the layout symbol
+	 */
+	VMINLINE ffi_type*
+	mapPrimitiveLayoutToFFIType(char layoutSymb)
+	{
+		ffi_type* typeFFI = NULL;
+
+		switch (layoutSymb) {
+		case 'V':	/* VOID */
+			typeFFI = &ffi_type_void;
+			break;
+		case 'C':	/* C_CHAR */
+			typeFFI = &ffi_type_sint8;
+			break;
+		case 'S':	/* C_SHORT */
+			typeFFI = &ffi_type_sint16;
+			break;
+		case 'I':	/* C_INT */
+			typeFFI = &ffi_type_sint32;
+			break;
+		case 'L':	/* either C_LONG or C_LONG_LONG */
+			typeFFI = &ffi_type_sint64;
```

Comment:
What does L even mean for passing out to pure C code?

Reply:
We use  `L` (8 bytes) to stands for `long` (`C_LONG`) on non-Windows platforms or `C_LONG_LONG` only on Windows. I added comment there for explanation.

Reply:
How does this differ from J?

Reply:
The code there is to parse/identify the argument type base with the 1st letter of the extract layout string of function descriptor (specified in the new feature).  e.g. The long type corresponds to `b64[abi/kind=LONG]` on Linux and `b64[abi/kind=LONG_LONG]` on Windows, in which case `L` represents the long type.

We are able to change to `J` by replacing the layout string at Java level but there is no need to to do given we have to match exactly what happens to the layout string at this point.

Reply:
Is this based on the RI? This is a senseless inconsistency given that all the other characters match java (with the exception of P).

Reply:
This is based on the API Spec at https://docs.oracle.com/en/java/javase/16/docs/api/jdk.incubator.foreign/jdk/incubator/foreign/CLinker.html#C_LONG, https://docs.oracle.com/en/java/javase/16/docs/api/jdk.incubator.foreign/jdk/incubator/foreign/CLinker.html#C_LONG_LONG as well as https://github.com/openjdk/panama-foreign/blob/b068d7d4041bb66ddef1f0a53a35f3cebf009026/doc/panama_ffi.md.

Reply:
Very annoying (though in retrospect, it was silly to choose L for objects and J for long, the opposite would have made infinitely more sense).

Reply:
We will need to preprocess the layout string of struct at phase 2 for easy handling in native due to complexity of struct string, in which case we are able to convert the layout string to a single letter (e.g. `LONG` or `LONG_LONG` maps to `J` for the convention in VM) if you prefer to do so.